File: CS/MyCardView/MyCardHandler.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using DevExpress.XtraGrid.Views.Grid.Handler;
using DevExpress.XtraGrid.Views.Grid;
using System.Windows.Forms;
using DevExpress.XtraGrid.Views.Card.Handler;
using DevExpress.XtraGrid.Views.Card;

namespace MyCardView
{
    public class MyCardHandler : CardHandler
    {
        public MyCardHandler(CardView cv)
            : base(cv)
        {
        }
        protected override void OnKeyDown(KeyEventArgs e)
        {
            base.OnKeyDown(e);
            if (e.KeyData == Keys.Delete && View.State == CardState.Normal)
                View.DeleteRow(View.FocusedRowHandle);
        }
    }
}

File: VB/MyCardView/MyCardHandler.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports System.Threading.Tasks
Imports DevExpress.XtraGrid.Views.Grid.Handler
Imports DevExpress.XtraGrid.Views.Grid
Imports System.Windows.Forms
Imports DevExpress.XtraGrid.Views.Card.Handler
Imports DevExpress.XtraGrid.Views.Card

Namespace MyCardView
	Public Class MyCardHandler
		Inherits CardHandler

		Public Sub New(ByVal cv As CardView)
			MyBase.New(cv)
		End Sub
		Protected Overrides Sub OnKeyDown(ByVal e As KeyEventArgs)
			MyBase.OnKeyDown(e)
			If e.KeyData = Keys.Delete AndAlso View.State = CardState.Normal Then
				View.DeleteRow(View.FocusedRowHandle)
			End If
		End Sub
	End Class
End Namespace
